asf_heat_pump_suitability/analysis/hn_zones/comparison_of_hn_zones.py

```diff
@@ -0,0 +1,350 @@
+        else 0.0
+    )
+    return avg_hn_score_pilot
+
+
+def calculate_mae_for_pilot_score(
+    hp_suitability_scores_with_desnz: pl.DataFrame, score_value: int
+) -> float:
+    """
+    Calculate and log the Mean Absolute Error (MAE) for entries with a specified 'DESNZ_pilot_score'.
+
+    Args:
+        hp_suitability_scores_with_desnz (pl.DataFrame): DataFrame containing 'DESNZ_pilot_score' and 'absolute_error' columns.
+        score_value (int): The value of 'DESNZ_pilot_score' (must be 1 or 0) for which to calculate the MAE.
+
+    Returns:
+        float: The MAE for the specified 'DESNZ_pilot_score'.
+
+    Raises:
+        ValueError: If score_value is not 1 or 0.
+    """
+    # Validate that score_value is either 1 or 0
+    if score_value not in {0, 1}:
+        raise ValueError("score_value must be either 1 or 0")
+
+    mae = hp_suitability_scores_with_desnz.filter(
+        pl.col("DESNZ_pilot_score") == score_value
+    )["absolute_error"].mean()
+
+    logging.info(
+        f"Mean Absolute Error (MAE) for DESNZ_pilot_score = {score_value}: {mae}"
+    )
+    return mae
+
+
+def calculate_mae_for_all(
+    hp_suitability_scores: pl.DataFrame, desnz_col: str, nesta_hn_score_col: str
+) -> Tuple[pl.DataFrame, float]:
+    """
+    Calculate the Mean Absolute Error (MAE) between the DESNZ pilot score and Nesta's HN suitability score, add the absolute error column to the DataFrame,
+    and log the result.
+
+    Args:
+        hp_suitability_scores_with_desnz (pl.DataFrame): DataFrame containing the DESNZ pilot (actual) and Nesta HN suitability score (predicted) columns.
+        desnz_col (str): Name of the column with DESNZ pilot (actual) values.
+        nesta_hn_score_col (str): Name of the column with Nesta HN suitability (predicted) values.
+
+    Returns:
+        Tuple[pl.DataFrame, float]:
+            - Updated DataFrame with 'absolute_error' column added.
+            - The Mean Absolute Error (MAE) between the actual and predicted columns.
+    """
+    # Calculate absolute error and add it as a new column
+    hp_suitability_scores_with_error = hp_suitability_scores.with_columns(
+        (pl.col(desnz_col) - pl.col(nesta_hn_score_col)).abs().alias("absolute_error")
+    )
+
+    # Calculate the mean of the absolute error
+    mae = hp_suitability_scores_with_error["absolute_error"].mean()
+    logging.info(
+        f"Mean Absolute Error (MAE) for {desnz_col} vs {nesta_hn_score_col}: {mae}"
+    )
+
+    return hp_suitability_scores_with_error, mae
+
+
+def main():
+    # Define the output directory and set up logging
+    output_dir = os.path.join(
+        PROJECT_DIR, "asf_heat_pump_suitability/analysis/hn_zones/output_data/"
+    )
+    setup_logging_and_file_path(output_dir)
+
+    # Load the data and perform spatial join
+    liverpool_with_desnz_hn_lsoa, list_of_liverpool_desnz_hn_lsoas = load_hn_geodata(
+        LIVERPOOL_GPKG_PATH, LSOA_SHP_PATH
+    )
```

Comment:
I think it's good practice to name your arguments in custom functions as an extra check that params are input in the correct order :) I would add those here and below.